doc/preface.chapter.md

```diff
@@ -6,11 +6,14 @@ The Nix Packages collection (Nixpkgs) is a set of thousands of packages for the
 Packages are available for several platforms, and can be used with the Nix
 package manager on most GNU/Linux distributions as well as [NixOS](https://nixos.org/nixos).
 
-This manual primarily describes how to write packages for the Nix Packages collection
-(Nixpkgs). Thus it’s mainly for packagers and developers who want to add packages to
-Nixpkgs. If you like to learn more about the Nix package manager and the Nix
-expression language, then you are kindly referred to the [Nix manual](https://nixos.org/nix/manual/).
-The NixOS distribution is documented in the [NixOS manual](https://nixos.org/nixos/manual/).
+This document is the user [_reference_](https://nix.dev/contributing/documentation/diataxis#reference) manual for Nixpkgs.
+Its purpose is to contain consise and complete listings of all functions, arguments and attributes with examples and cross-links.
```

Comment:
What about behavior?

```suggestion
Its purpose is to contain consise and complete listings of all functions, arguments, attributes and behaviors with examples and cross-links.
```

Reply:
```suggestion
Its purpose is to describe the Nixpkgs machinery, in a concise and orderly manner, the entire public interface and all relevant behaviors, with examples and cross links.
```

I don' think we have to list out what kinds of things the public interface may consist of. What's important here is completeness.
I'm including behaviors because we have a bunch of things that aren't purely functional, such as `trace`, `setup.sh`, and all the hooks.

Reply:
A preface seems particularly prone to being overlooked. Perhaps we can focus attention on the relevant keywords.

```suggestion
Its purpose is to describe the **Nixpkgs machinery**, in a **concise** and orderly manner, the **entire** public interface and all relevant behaviors, with examples and cross links.
```

Reply:
I like the suggestion, except "Nixpkgs machinery".

Reply:
Agreeing with @fricklerhandwerk's final suggestion, applying!